privacy_evaluator/attacks/property_inference_attack.py

```diff
@@ -0,0 +1,340 @@
+from privacy_evaluator.attacks.attack import Attack
+from privacy_evaluator.classifiers.classifier import Classifier
+from privacy_evaluator.models.train_cifar10_torch import data, train
+
+import math
+import numpy as np
+import torch
+import tensorflow as tf
+from sklearn.svm import SVC
+from art.attacks.evasion import FastGradientMethod
+from art.estimators.classification import SklearnClassifier
+from typing import Tuple, Any, Dict, List
+
+
+class PropertyInferenceAttack(Attack):
+    def __init__(self, target_model: Classifier):
+        """
+        Initialize the Property Inference Attack Class.
+        :param target_model: the target model to be attacked
+        """
+
+        super().__init__(target_model, None, None, None, None)
+
+    def create_shadow_training_set(
+        self,
+        dataset: torch.utils.data.Dataset,
+        amount_sets: int,
+        size_set: int,
+        property_num_elements_per_classes: Dict[int, int],
+    ) -> Tuple[
+        List[torch.utils.data.Dataset],
+        List[torch.utils.data.Dataset],
+        Dict[int, int],
+        Dict[int, int],
+    ]:
+        """
+        Create the shadow training sets, half fulfill the property, half fulfill the negation of the property.
+        The function works for the specific binary case that the property is a fixed distribution specified in the input
+        and the negation of the property is a balanced distribution.
+        :param dataset: Dataset out of which shadow training sets should be created
+        :param amount_sets: how many shadow training sets should be created
+        :param size_set: size of one shadow training set for one shadow classifier
+        :param property_num_elements_per_classes: number of elements per class, this is the property
+        :return: shadow training sets for property,
```

Comment:
Please add also the types INSIDE the dictionary.

Reply:
We discussed this in the sprint review, the types inside the dictionary were already given.